src/coreclr/tools/Common/TypeSystem/Common/TypeDesc.cs

```diff
@@ -676,6 +676,17 @@ public bool IsByRefLike
             }
         }
 
+        /// <summary>
+        /// Gets a value indicating whether this is a value array type
+        /// </summary>
+        public bool IsInlineArray
```

Comment:
Can you move this to `MetadataType`? This is a similar question to `IsExplicitLayout`, etc. and cannot be answered without metadata (we cannot compute the right answer for this in the NativeAOT runtime type system).

All the places that need this de facto already operate on `MetadataType` - they just need a cast (e.g. `GatherClassGCLayout` operates on `TypeDesc`, but it will only ever be called with valuetypes and those can always be safely cast to `MetadataType` and the GC layout algorithm operates on `DefType`, but should operate on `MetadataType` instead - the difference between DefType and MetadataType only matters outside the compiler - in the compiler they're the same thing).

Reply:
>All the places that need this de facto already operate on MetadataType

That does not seem to be a case for `InstantiatedType`

Reply:
`ComputeTypeFlags` defers to the underlying `_typeDef` and that is not a `MetadataType`  when `TYPE_LOADER_IMPLEMENTATION` is set.

Maybe I should just assume in it not an inline array in such case?

How does it work with `IsSequentialLayout` ?

Reply:
> That does not seem to be a case for `InstantiatedType`

* Place the override into InstantiatedType.Metadata.cs so that the overrides are only in effect when `InstantiatedType` derives from `MetadataType`.
* Handle the flag the same as we do for `AddComputedIntrinsicFlag` (put the partial method into InstantiatedType.Metadata.cs).